### PR TITLE
bugfix: prefer using Testem.afterTests instead of QUnit.done

### PR DESCRIPTION
**Pull request by fivetanley**
This affects folks using `setupMiddlewareReporter`.

Sometimes, like in CI environments, there can exist a race condition where Testem closes before the local middleware reporter can finish writing the reports to the filesystem.

According to [Testem's README][1], it appears the proper way to do this is to use `Testem.afterTests` to send reports. This will keep testem open and prevent the process from exiting before the reports are written.

This seems to be a [known issue with Testem's QUnit integration][2].

[1]: https://github.com/testem/testem/blob/0bd05cec7f83bfe9f2e10ddfab54556d724ee179/README.md#running-browser-code-after-tests-complete
[2]: https://github.com/testem/testem/issues/1577

### CODE REVIEW

**Comment by fivetanley** [comment]
needs https://github.com/ember-a11y/ember-a11y-testing/pull/559 for typescript CI failure checks

**Comment by MelSumner** [comment]
@fivetanley thanks for the PR. I approved the first PR; I'll review this one and do a release in the morning. 👍

File: addon-test-support/setup-middleware-reporter.ts [hunk 1]
```diff
@@ -149,15 +159,28 @@ export function setupMiddlewareReporter() {
...
+  if (window.Testem) {
+    window.Testem.afterTests(async function (_config, _data, callback) {
+      try {
+        await sendViolationsToServer();
+      } finally {
+        callback();
+      }
+    });
+  } else {
+    QUnit.done(async function () {
```
Comment (fivetanley) [inline addon-test-support/setup-middleware-reporter.ts:171]: falling back to QUnit.done in the rare case that folks are running in an environment other than Testem. I remembered some teams save their runners on S3 or other file stores for debugging purposes.

Comment (josephdsumner) [reply addon-test-support/setup-middleware-reporter.ts:171]: The fallback here is appreciated as a review and consumer!

**Review by josephdsumner — APPROVED** [review]

**Review by josephdsumner — APPROVED** [review]

**Review by MelSumner — APPROVED** [review]
Thanks for figuring this out! 

Noting for myself and other reviewers/users/watchers that we'll need to figure out why the `ember-beta` scenario is failing rather soon. 👀

**Pull request merged**